src/service/video-manager-impl.js

```diff
@@ -436,6 +540,136 @@ class VideoEntry {
   }
 
   /**
+   * Called when the video's position in the viewport changed.
+   * @param {PositionInViewportEntryDef} newPos
+   * @private
+   */
+  onPositionChanged_(newPos) {
+
+    const vidElement = scopedQuerySelector(this.video.element, 'video, iframe');
+    const vidStyle = vidElement.style;
+
+    // How much to scale the video by when minimized
+    const SCALE = 0.6;
+
+    // Temporary fix until PositionObserver somehow tracks objects outside of
+    // the viewport
+    if (!newPos.positionRect
+       && this.getPlayingState() == PlayingStates.PLAYING_MANUAL
+       && this.minimizePosition_ != MinimizePositions.DEFAULT) {
+      // Hide the controls.
+      this.video.hideControls();
+      vidElement.classList.add('i-amphtml-dockable-video-minimizing');
+      vidStyle.height = SCALE * this.initialHeight_ + 'px';
```

Comment:
let's try to only use `transform: scale` and `transform-origin` that should remove the need to know initial width/height/left as well.